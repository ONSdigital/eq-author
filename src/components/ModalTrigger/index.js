import React from "react";
import PropTypes from "prop-types";

class ModalTrigger extends React.Component {
  static propTypes = {
    children: PropTypes.func.isRequired
  };

  state = {
    isOpen: false
  };

  handleModalOpen = () => {
    this.setState({ isOpen: true });
  };

  handleModalClose = () => {
    this.setState({ isOpen: false });
  };

  render() {
    return this.props.children({
      isOpen: this.state.isOpen,
      onOpen: this.handleModalOpen,
      onClose: this.handleModalClose
    });
  }
}

export default ModalTrigger;
