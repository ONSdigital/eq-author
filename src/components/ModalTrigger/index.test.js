import React from "react";
import { mount } from "enzyme";
import ModalTrigger from "./index";

describe("ModalTrigger", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <ModalTrigger>
        {({ isOpen, onClose, onOpen }) => (
          <React.Fragment>
            <button id="open" onClick={onOpen}>
              Open
            </button>
            <button id="close" onClick={onClose}>
              Close
            </button>
            <input type="checkbox" checked={isOpen} />
          </React.Fragment>
        )}
      </ModalTrigger>
    );
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should initially be closed", () => {
    expect(wrapper.find("input").prop("checked")).toBe(false);
  });

  it("should allow modal to be opened", () => {
    wrapper.find("#open").simulate("click");
    expect(wrapper.find("input").prop("checked")).toBe(true);
  });

  it("should allow for modal to be closed", () => {
    wrapper.find("#open").simulate("click");
    expect(wrapper.find("input").prop("checked")).toBe(true);

    wrapper.find("#close").simulate("click");
    expect(wrapper.find("input").prop("checked")).toBe(false);
  });
});
