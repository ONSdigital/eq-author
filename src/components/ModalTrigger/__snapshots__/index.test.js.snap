// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalTrigger should render 1`] = `
<ModalTrigger>
  <button
    id="open"
    onClick={[Function]}
  >
    Open
  </button>
  <button
    id="close"
    onClick={[Function]}
  >
    Close
  </button>
  <input
    checked={false}
    type="checkbox"
  />
</ModalTrigger>
`;
